### PR TITLE
[SYCL] Changed sycl::backend::level0 to sycl::backend::level_zero

### PR DESCRIPTION
**Pull request by glyons-intel**
Signed-off-by: Gail Lyons <gail.lyons@intel.com>

### CODE REVIEW

File: sycl/source/detail/config.hpp [hunk 1]
```diff
@@ -120,7 +120,7 @@ template <> class SYCLConfig<SYCL_BE> {
     const char *ValStr = BaseT::getRawValue();
     const std::array<std::pair<std::string, backend>, 3> SyclBeMap = {
         {{"PI_OPENCL", backend::opencl},
-         {"PI_LEVEL0", backend::level0},
+         {"PI_LEVEL0", backend::level_zero},
```
Comment (kbobrovs) [inline sycl/source/detail/config.hpp:123]:
```suggestion
         {"PI_LEVEL_ZERO", backend::level_zero},
```

Comment (kbobrovs) [reply sycl/source/detail/config.hpp:123]: also docs which specify env variables need to be updated

File: sycl/source/detail/pi.cpp [hunk 2]
```diff
@@ -215,7 +215,7 @@ bool findPlugins(vector_class<std::pair<std::string, backend>> &PluginNames) {
   // env only.
   //
   PluginNames.emplace_back(OPENCL_PLUGIN_NAME, backend::opencl);
-  PluginNames.emplace_back(LEVEL0_PLUGIN_NAME, backend::level0);
+  PluginNames.emplace_back(LEVEL0_PLUGIN_NAME, backend::level_zero);
```
Comment (smaslov-intel) [inline sycl/source/detail/pi.cpp:218]: LEVEL0_PLUGIN_NAME -> LEVEL_ZERO_PLUGIN_NAME

File: sycl/source/detail/program_manager/program_manager.cpp [hunk 3]
```diff
@@ -766,7 +766,7 @@ ProgramManager::build(ProgramPtr Program, const ContextImplPtr Context,
   // is built during piProgramCreate.
   // TODO: remove this check as soon as piProgramCompile/piProgramLink will be
   // implemented in L0 plugin.
```
Comment (smaslov-intel) [inline sycl/source/detail/program_manager/program_manager.cpp:768]: L0 -> Level-Zero, and please grep for other instances of "L0"

**Comment by smaslov-intel** [comment]
There are other instances of "Level0" and "L0" that need to be renamed, e.g. here:
~~~
CL/sycl/detail/pi.hpp:#define LEVEL0_PLUGIN_NAME "pi_level0.dll"
CL/sycl/detail/pi.hpp:#define LEVEL0_PLUGIN_NAME "libpi_level0.so"
~~~

File: sycl/source/detail/config.hpp (hunk 1 above)
Comment (glyons-intel) [reply sycl/source/detail/config.hpp]: Done.

File: sycl/source/detail/pi.cpp (hunk 2 above)
Comment (glyons-intel) [reply sycl/source/detail/pi.cpp]: Done.

File: sycl/source/detail/program_manager/program_manager.cpp (hunk 3 above)
Comment (glyons-intel) [reply sycl/source/detail/program_manager/program_manager.cpp]: Done.

File: sycl/plugins/level_zero/CMakeLists.txt [hunk 4]
```diff
@@ -66,44 +66,44 @@ target_include_directories(L0Loader-Headers
...
 
   # Filter symbols based on the scope defined in the script file,
   # and export pi* function symbols in the library.
-  target_link_libraries( pi_level0
+  target_link_libraries( pi_level_zero
     PRIVATE "-Wl,--version-script=${linker_script}"
   )
 endif()
 
 if (TARGET l0-loader)
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/CMakeLists.txt:95]: "l0"

File: sycl/plugins/level_zero/CMakeLists.txt [hunk 5]
```diff
@@ -66,44 +66,44 @@ target_include_directories(L0Loader-Headers
...
-  add_dependencies(pi_level0 l0-loader)
+  add_dependencies(pi_level_zero l0-loader)
 endif()
-  add_dependencies(sycl-toolchain pi_level0)
+  add_dependencies(sycl-toolchain pi_level_zero)
 
-  target_link_libraries(pi_level0 PRIVATE "${L0_LOADER}")
+  target_link_libraries(pi_level_zero PRIVATE "${L0_LOADER}")
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/CMakeLists.txt:100]: "L0"

File: sycl/source/detail/config.hpp [hunk 6]
```diff
@@ -120,17 +122,20 @@ template <> class SYCLConfig<SYCL_BE> {
     const char *ValStr = BaseT::getRawValue();
     const std::array<std::pair<std::string, backend>, 3> SyclBeMap = {
         {{"PI_OPENCL", backend::opencl},
-         {"PI_LEVEL0", backend::level_zero},
+         {"PI_LEVEL_ZERO", backend::level_zero},
          {"PI_CUDA", backend::cuda}}};
     if (ValStr) {
+      if (strcmp(ValStr, "PI_LEVEL0") == 0) {
```
Comment (smaslov-intel) [inline sycl/source/detail/config.hpp:128]: Why did you want to support "PI_LEVEL0" name? Is this to accommodate current users/tests? Please add it to the array  at line 123.

File: sycl/plugins/level_zero/CMakeLists.txt (hunk 4 above)
Comment (glyons-intel) [reply sycl/plugins/level_zero/CMakeLists.txt:95]: done.

File: sycl/plugins/level_zero/CMakeLists.txt (hunk 5 above)
Comment (glyons-intel) [reply sycl/plugins/level_zero/CMakeLists.txt:100]: done.

File: sycl/source/detail/config.hpp (hunk 6 above)
Comment (glyons-intel) [reply sycl/source/detail/config.hpp:128]: yes, for backwards compatibility.  done.

File: sycl/test/abi/sycl_symbols_linux.dump [hunk 7]
```diff
@@ -3724,7 +3728,9 @@ _ZN2cl4sycl6detail13MemoryManager13releaseMemObjESt10shared_ptrINS1_12context_im
 _ZN2cl4sycl6detail13MemoryManager16allocateMemImageESt10shared_ptrINS1_12context_implEEPNS1_11SYCLMemObjIEPvbmRK14_pi_image_descRK16_pi_image_formatRKS3_INS1_10event_implEERKS5_RP9_pi_event
 _ZN2cl4sycl6detail13MemoryManager17allocateMemBufferESt10shared_ptrINS1_12context_implEEPNS1_11SYCLMemObjIEPvbmRKS3_INS1_10event_implEERKS5_RP9_pi_event
 _ZN2cl4sycl6detail13MemoryManager18allocateHostMemoryEPNS1_11SYCLMemObjIEPvbm
+_ZN2cl4sycl6detail13MemoryManager18releaseImageBufferESt10shared_ptrINS1_12context_implEEPv
```
Comment (bader) [inline sycl/test/abi/sycl_symbols_linux.dump:3731]: I guess we should bump SYCL library version.

Comment (glyons-intel) [reply sycl/test/abi/sycl_symbols_linux.dump:3738]: Done.

File: sycl/source/detail/config.hpp [hunk 8]
```diff
@@ -15,6 +15,7 @@
 #include <algorithm>
 #include <array>
 #include <cstdlib>
+#include <cstring>
```
Comment (bader) [inline sycl/source/detail/config.hpp:18]:
```suggestion
#include <string>
```
I don't see usages on `cstring` functions, but `std::string` is used here. Should `string` be included instead?

File: sycl/test/basic_tests/image_accessor_readsampler.cpp [hunk 9]
```diff
@@ -6,6 +6,7 @@
 // RUN: %CPU_RUN_PLACEHOLDER %t.out
 // RUN: %GPU_RUN_PLACEHOLDER %t.out
 //
+
```
Comment (bader) [inline sycl/test/basic_tests/image_accessor_readsampler.cpp:9]: Do we need two empty lines here?

File: sycl/test/spec_const/spec_const_hw.cpp [hunk 10]
```diff
@@ -1,9 +1,10 @@
-// UNSUPPORTED: cuda || level0
+// UNSUPPORTED: cuda || level_zero
 //
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
 // RUN: env SYCL_DEVICE_TYPE=HOST %t.out
 // RUN: %CPU_RUN_PLACEHOLDER %t.out
 // RUN: %GPU_RUN_PLACEHOLDER %t.out
+
```
Comment (bader) [inline sycl/test/spec_const/spec_const_hw.cpp:7]: Do we need two empty lines here?

File: sycl/test/spec_const/spec_const_redefine.cpp [hunk 11]
```diff
@@ -1,9 +1,10 @@
-// UNSUPPORTED: cuda || level0
+// UNSUPPORTED: cuda || level_zero
 //
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
 // RUN: env SYCL_DEVICE_TYPE=HOST %t.out
 // RUN: env SYCL_PI_TRACE=2 %CPU_RUN_PLACEHOLDER %t.out 2>&1 %CPU_CHECK_PLACEHOLDER
 // RUN: env SYCL_PI_TRACE=2 %GPU_RUN_PLACEHOLDER %t.out 2>&1 %GPU_CHECK_PLACEHOLDER
+
```
Comment (bader) [inline sycl/test/spec_const/spec_const_redefine.cpp:7]: Do we need two empty lines here?

File: sycl/source/detail/config.hpp (hunk 8 above)
Comment (glyons-intel) [reply sycl/source/detail/config.hpp]: done.

File: sycl/test/basic_tests/image_accessor_readsampler.cpp (hunk 9 above)
Comment (glyons-intel) [reply sycl/test/basic_tests/image_accessor_readsampler.cpp]: done

File: sycl/test/spec_const/spec_const_hw.cpp (hunk 10 above)
Comment (glyons-intel) [reply sycl/test/spec_const/spec_const_hw.cpp]: done

File: sycl/test/spec_const/spec_const_redefine.cpp (hunk 11 above)
Comment (glyons-intel) [reply sycl/test/spec_const/spec_const_redefine.cpp]: done

**Comment by bader** [comment]
@glyons-intel, it looks like there were other commits to the sycl branch, which conflict with this PR. Could you resolve conflicts, please?

**Pull request merged**